onset_maproom/calc.py

```diff
@@ -33,6 +35,80 @@ def onset_date(daily_rain, early_start_day, early_start_month, search_days, rain
 #  onset_date = onset_date[time_coord] + onset_date  
   return onset_date
 
+# Time functions
+
+def daily_groupby_season(daily_data, start_day, start_month, end_day, end_month, time_coord="T"):
+  """Groups daily data by yearly seasons from day-month edges of season
+  Can then apply functions on each group to create yearly time series of seasonal quantities
+  If starting day-month is 29-Feb, uses 1-Mar.
```

Comment:
I think a more natural interpretation would be that the season starts on Feb 29 in leap years and Mar 1 in non-leap years. Not terribly important; I doubt anyone will actually do this.

Reply:
Currently it's not what I am doing. If a user gives a start of 29 Feb, I make it 1 Mar. I am dealing with the end differently because it makes sense to be able to select the entire month of Feb, so if the end input is 28 Feb, they get seasons that end 28 Feb, if they enter 29 Feb, they get the full month of Feb. I feel like that might be useful especially if later this kind of function dives into a bigger environment where "Feb" selects the full month of February, for instance. Whereas I feel like someone who wants a season that starts with 29 Feb when it exists and 1 Mar when it doesn't, would be a jerk :) But obviously I am biased in those choices I make and maybe it would be more natural to deal with start symmetrically to how I deal with end...

Reply:
I understand, the comment correctly describes the implementation, but I'm suggesting that the implementation decision is a little bit surprising/asymmetrical.

Apparently I have to "resolve" this conversation before the PR can be merged, so I'm going to do that, even though the issue isn't really resolved. It's just not important enough to block us from moving ahead.